### ISSUE TITLE
Bug with Bi-directional pins?

### ISSUE THREAD

**Issue by mitolegy** [hneemann/Digital#511]
there appears to be a bug in the Input component. in previous versions. turning on the settings "is three-state input" and "no zero output" on inputs has allowed for bidirectional data flow through the pin. however, in the latest version. this no longer seems to be true. i kept getting short circuit errors when facing two input components. as if they did not have the two options enabled. 

![image](https://user-images.githubusercontent.com/5740009/92950673-59908200-f45d-11ea-9c6d-cc40100c7579.png)
basically, turning on these
![image](https://user-images.githubusercontent.com/5740009/92950726-7036d900-f45d-11ea-8811-46d535d3493c.png)
used to let me do this.

**Comment by hneemann** [comment]
It is more a consitency problem. You also have to set at least one of the inpuits to a default value of Z. Otherwise it could cause a error while starting up the simulation.

But while experimenting to reproduce the bug i found that there is another problem: The default value is sometimes set to "?" which is not allowed. In this case you have to reset the value to "Z".

**Comment by hneemann** [comment]
fixed in a526c91bc9a407acd4a764cfccdb94e75d06010f
